conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/GramineEnclaveEnvironment.kt

```diff
@@ -9,10 +9,11 @@ import java.security.MessageDigest
 /**
  * This class is the enclave environment intended for use with gramine-direct.
  */
-class GramineDirectEnclaveEnvironment(
-        enclaveClass: Class<*>,
-        override val hostInterface: SocketEnclaveHostInterface,
-        private val mrsigner: ByteArray
+class GramineEnclaveEnvironment(
+    enclaveClass: Class<*>,
+    override val hostInterface: SocketEnclaveHostInterface,
+    private val simulationMrSigner: ByteArray,
+    override val enclaveMode: EnclaveMode
```

Comment:
Add a `requires` check that this isn't mock.

Reply:
We have that just before calling such a constructor. You think it is really required?
```
        validateEnclaveMode()
        val env = GramineEnclaveEnvironment(enclaveClass, hostInterface, simulationMrSigner!!, enclaveMode)
```

Reply:
Yes, becuase the class should do its own checking for any preconditions it has.